tools/img_split_dota.py

```diff
@@ -0,0 +1,363 @@
+               'basketball-court', 'storage-tank', 'soccer-ball-field',
+               'roundabout', 'harbor', 'swimming-pool', 'helicopter',
+               'container-crane', 'airport', 'helipad']
+
+def add_parser(parser):
+    #argument for processing
+    parser.add_argument('--base_json', type=str, default=None,
+                        help='json config file for split images')
+    parser.add_argument('--nproc', type=int, default=20,
+                        help='the procession number')
+
+    #argument for loading data
+    parser.add_argument('--load_type', type=str, default=None,
+                        help='loading function type')
+    parser.add_argument('--img_dirs', nargs='+', type=str, default=None,
+                        help='images dirs, must give a value')
+    parser.add_argument('--ann_dirs', nargs='+', type=str, default=None,
+                        help='annotations dirs, optional')
+    parser.add_argument('--classes', type=str, default=None,
+                        help='the classes for loading data')
+    parser.add_argument('--prior_annfile', type=str, default=None,
+                        help='prior annotations merge to data')
+    parser.add_argument('--merge_type', type=str, default='addition',
+                        help='prior annotations merging method')
+
+    #argument for splitting image
+    parser.add_argument('--sizes', nargs='+', type=int, default=[1024],
+                        help='the sizes of sliding windows')
+    parser.add_argument('--gaps', nargs='+', type=int, default=[512],
+                        help='the steps of sliding widnows')
+    parser.add_argument('--rates', nargs='+', type=float, default=[1.],
+                        help='same as DOTA devkit rate, but only change windows size')
+    parser.add_argument('--img_rate_thr', type=float, default=0.6,
+                        help='the minimal rate of image in window and window')
+    parser.add_argument('--iof_thr', type=float, default=0.7,
+                        help='the minimal iof between a object and a window')
+    parser.add_argument('--no_padding', action='store_true',
+                        help='not padding patches to regular size')
+    parser.add_argument('--padding_value', nargs='+',type=int, default=[0],
+                        help='padding value, 1 or channel number')
+
+    #argument for saving
+    parser.add_argument('--filter_empty', action='store_true',
+                        help='filter out empty patches, speed up trining splitting')
+    parser.add_argument('--save_dir', type=str, default='.',
+                        help='to save pkl and splitted images')
+    parser.add_argument('--save_ext', type=str, default='.png',
+                        help='the extension of saving images')
+    parser.add_argument('--generate_txt', action='store_true', default=True,
+                        help='to generate txt for splitted images')
```

Comment:
It's better to not set default value for the `store_true` argument

Reply:
ok